# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1292295595 18000
#      Mon Dec 13 21:59:55 2010 -0500
# Node ID f8cc86b1503ebfdde3eb2645cac27b6b066225e8
# Parent  34c5cd5a17ecda5d093ccc6f34daaeaa653eb91a
__print_parse_opts__.m: Fix indexing bug for varargin.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-12-13  Ben Abbott  <bpabbott@mac.com>
+
+	* plot/__print_parse_opts__.m: Fix indexing bug for varargin.
+
 2010-12-13  Rik  <octave@nomad.inbox5.com>
 
 	* statistics/distributions/betapdf.m: Add test for large values of
 	alpha and beta.
 
 2010-12-13  Rik  <octave@nomad.inbox5.com>
 
 	* strings/strmatch.m: Add test with null search pattern.
diff --git a/scripts/plot/__print_parse_opts__.m b/scripts/plot/__print_parse_opts__.m
--- a/scripts/plot/__print_parse_opts__.m
+++ b/scripts/plot/__print_parse_opts__.m
@@ -71,17 +71,17 @@ function arg_st = __print_parse_opts__ (
   endif
   arg_st.unlink = {};
   
   if (nargin > 0 && isfigure (varargin{1}))
     arg_st.figure = varargin{1};
     varargin(1) = [];
   endif
 
-  for i = 1:nargin
+  for i = 1:numel(varargin)
     arg = strtrim (varargin{i});
     if (ischar (arg))
       if (strcmp (arg, "-color"))
         arg_st.use_color = 1;
       elseif (strcmp (arg, "-append"))
         arg_st.append_to_file = true;
       elseif (strcmp (arg, "-mono"))
         arg_st.use_color = -1;
